perllib/FixMyStreet/App/Model/PhotoSet.pm

```diff
@@ -416,4 +416,31 @@ sub redact_image {
     return $new_set;
 }
 
+# Repeatedly shrink any images over the given size to the given percentage
+# of their original size until they are small enough.
+# Returns the new photoset and a bool indicating if any images were shrunk.
+sub shrink_all_to_size {
+    my ($self, $size_bytes, $resize_percent) = @_;
+
+    my $shrunk = 0;
+    my @images = $self->all_ids;
+    foreach my $i (0.. $#images) {
+        my $blob = $self->get_raw_image($i)->{data};
+        while (length $blob > $size_bytes) {
```

Comment:
Could we multiple $resize_percent by itself each loop, and always shrink from the full size image? Just worried that repeated shrinkings might degrade the quality of the image as opposed to one shrinking (of whatever %).